packages/dcos-integration-test/extra/test_networking.py

```diff
@@ -178,13 +226,52 @@ def setup_vip_workload_tests(dcos_api_session, container, vip_net, proxy_net):
     # a fresh state, but in this case the previous app load may still be winding down,
     # so allow a larger buffer time
     retry_on_result=lambda res: res is None)
-def wait_for_tasks_healthy(dcos_api_session, app_definition):
-    proxy_info = dcos_api_session.marathon.get('v2/apps/{}'.format(app_definition['id'])).json()
-    if proxy_info['app']['tasksHealthy'] == app_definition['instances']:
-        return proxy_info
+def wait_for_tasks_healthy(dcos_api_session, pod, app_definition):
```

Comment:
I would prefer just making a `wait_for_app_tasks_healthy` and `wait_for_pod_tasks_healthy` rather than add an `if` in such that the only shared code is the retry decorator